### ISSUE TITLE
GM under-segmented on Phillips data in data-multi-subject

### ISSUE THREAD

**Issue by renelabounek** [spine-generic/spine-generic#273]
Hello SCT friends,
has anything changed in Phillips data or GM segmentation process over database or SCT versions?

Since Friday, I have updated data-multi-subject and SCT to the latest versions and GM sementation results differs a lot for Phillips data. Everything measured in C3/4 segment.

Current vesrions:
SCT version: git-master-c7a8072fd63a06a2775a74029c042833f0fce510
data-multi-subject: r20230223-3-g901964b

First see graphs for previous older SCT and database versions (generated in summer/fall 2022).
![fig1](https://github.com/spine-generic/spine-generic/assets/12597538/365794bb-ea47-4765-82e8-a5017812a291)

And, now look at current CSA-GM measurements for Philips data with the latest SCT and database versions:
![fig_corr_body_csa](https://github.com/spine-generic/spine-generic/assets/12597538/224e9f57-9bc8-4d4f-a061-4f701a4809b5)

Note that neither CSA-SC and CSA-WM appear to be changed substantially, nor CSA-GM for Siemens and GE scans. Data were processed with this version of [proces_data.sh](https://github.com/renelabounek/spine-generic/blob/rl/revise1-height-weight-analysis/process_data.sh) 

Here is a list of Phillips scans where I get C3/4 CSA-GM < 8mm^2 after database and SCT update.
 - sub-fslAchieva01
 - sub-nottwil01
 - sub-tokyoIngenia04
 - sub-vuiisAchieva01
 - sub-vuiisIngenia03

Here is example of obtained GM segmentation for sub-vuiisAchieva01 providing CSA-GM = 2.58 mm^2. I am not saying that the GM contrast is the best in the scan, however, the GM segmentation is obviously very off.
![sub-vuiisAchieva01_gm_segmentation](https://github.com/spine-generic/spine-generic/assets/12597538/b571bf28-e79b-452a-80ed-c8269f4d5c98)

Please, any idea for an easy fix of GM segmentation on the Phillips data? :-) 

Best,
Rene

**Comment by valosekj** [comment]
Hey @renelabounek!

Thanks for pointing this out!

> Here is example of obtained GM segmentation for sub-vuiisAchieva01 providing CSA-GM = 2.58 mm^2. I am not saying that the GM contrast is the best in the scan, however, the GM segmentation is obviously very off.
![sub-vuiisAchieva01_gm_segmentation](https://github.com/spine-generic/spine-generic/assets/12597538/b571bf28-e79b-452a-80ed-c8269f4d5c98)

This looks like an automatic segmentation, right? Why are you not using manually corrected GM seg located under `derivatives/labels` (for this particular subject [here](https://github.com/spine-generic/data-multi-subject/tree/master/derivatives/labels/sub-vuiisAchieva01/anat))?

`https://github.com/spine-generic/data-multi-subject/tree/master/derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_gmseg-manual.nii.gz`:

<img width="354" alt="image" src="https://github.com/spine-generic/spine-generic/assets/39456460/ab51bb6d-e26b-40b4-b86f-90a5bd3d098b">

Your script actually checks if manual GM seg exists (lines [here](https://github.com/renelabounek/spine-generic/blob/rl/revise1-height-weight-analysis/process_data.sh#L110-L116)) and uses it if so. Could you please check your log to see if the manual GM seg was found by your script?

**Comment by renelabounek** [comment]
I appears the script did not identified existing manual GM segmentation for this particular subject

+ segment_gm_if_does_not_exist sub-vuiisAchieva01_T2star_rms t2s
+ local file=sub-vuiisAchieva01_T2star_rms
+ local contrast=t2s
+ FILESEG=sub-vuiisAchieva01_T2star_rms_gmseg
+ FILESEGMANUAL=~/spine-generic/data-multi-subject/derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_rms_gmseg-manual.nii.gz
+ echo 'Looking for manual segmentation: ~/spine-generic/data-multi-subject/derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_rms_gmseg-manual.nii.gz'
Looking for manual segmentation: ~/spine-generic/data-multi-subject/derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_rms_gmseg-manual.nii.gz
+ [[ -e ~/spine-generic/data-multi-subject/derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_rms_gmseg-manual.nii.gz ]]
+ echo 'Not found. Proceeding with automatic segmentation.'
Not found. Proceeding with automatic segmentation.
+ sct_deepseg_gm -i sub-vuiisAchieva01_T2star_rms.nii.gz -qc ~/spine-generic/results/data-multi-subject/qc -qc-subject sub-vuiisAchieva01

--
Spinal Cord Toolbox (git-master-c7a8072fd63a06a2775a74029c042833f0fce510)

sct_deepseg_gm -i sub-vuiisAchieva01_T2star_rms.nii.gz -qc ~/spine-generic/results/data-multi-subject/qc -qc-subject sub-vuiisAchieva01
--


It may again be some problem with working on the forked directory. However, git annex outputs seems fine.

``
[user@sever data-multi-subject]$ git fetch upstream +refs/heads/git-annex:refs/remotes/upstream/git-annex
Auto packing the repository for optimum performance. You may also
run "git gc" manually. See "git help gc" for more information.
Counting objects: 19767, done.
Delta compression using up to 32 threads.
Compressing objects: 100% (18989/18989), done.
Writing objects: 100% (19767/19767), done.
Total 19767 (delta 5291), reused 0 (delta 0)
Checking connectivity: 223229, done.
[user@server data-multi-subject]$
``


``
[user@sever data-multi-subject]$ git pull && git annex get .
Already up-to-date.
get derivatives/labels/sub-beijingVerio04/anat/sub-beijingVerio04_T2star_seg-manual.nii.gz (from amazon...) 
ok                                
get derivatives/labels/sub-beijingVerio04/dwi/sub-beijingVerio04_rec-average_dwi_seg-manual.nii.gz (from amazon...) 
ok                                
(recording state in git...)
[user@sever data-multi-subject]$
``


Moreover, I see some version of the manual seg file exists on my HDD. However, the proces_data.sh script is looking for **sub-vuiisAchieva01_T2star_rms_gmseg-manual** file. And, I have on my HDD **sub-vuiisAchieva01_T2star_gmseg-manual** file, i.e. missing **rms** in file naming. Which leads us back to some issue with git annex on the forked directory.

``
[user@server data-multi-subject]$ ls derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_gmseg-manual.*
derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_gmseg-manual.json  derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_gmseg-manual.nii.gz
[user@server data-multi-subject]$
``

I will try to look if I find some further indices here: https://github.com/spine-generic/data-multi-subject/issues/125

**Comment by valosekj** [comment]
Sorry, I do not remember, what is the exact reason to work with the forked dataset? Wouldn't it be easier to `git clone` and `git annex` the latest release (you could also point to the release in the paper)? If the reason is this waiting PR (https://github.com/spine-generic/data-multi-subject/pull/148), sorry for that, it will be merged soon.

**Comment by renelabounek** [comment]
I forked the database, because that has been the only way how to edit the database on my site. I can clone the latest official relase to other  HDD location. however, I will have the data amount twice on the HDD.

I think the PR 148 is unrelated.

**Comment by renelabounek** [comment]
I found it, there is tweak eitther in the proces_data.sh script or the database.
The script is looking for the *_rms_* in the file name. However, neither the latest master file version on github does not posses the *_rms_* string in the filename.

https://github.com/spine-generic/data-multi-subject/blob/master/derivatives/labels/sub-vuiisAchieva01/anat/sub-vuiisAchieva01_T2star_seg-manual.nii.gz

for now, I will adjust the script to look for file without rms. Is this the case only in the T2star image or is the fix required over all images analzed in the process_data.sh?

**Comment by jcohenadad** [comment]
whatever fixes the issue, it would be great to push the fix to the main repos, so others can benefit from the fix. Thanks!

**Comment by valosekj** [comment]
Indeed, [process_data.sh](https://github.com/spine-generic/spine-generic/blob/master/process_data.sh) is currently out of sync with the latest release of data-multi-subject (r20230223: https://github.com/spine-generic/data-multi-subject/tree/r20230223). Issue created: https://github.com/spine-generic/spine-generic/issues/274
